Added de_DE localisation

mocdaniel opened the pull request:
I tried to translate to the shortest possible idiomatic German equivalents, some got rather long anyways, so maybe somebody would like to review this.

Does paru support localised input for [Y/n] dialogues and the likes? When translating, I pretended it did, this might need correction depending on paru's behaviour (e.g. see [J/n] translation).

Some rather technical terms such as `make`, `build`, `repository` and `maintainer` remain the same in German.

Morganamilo commented:
Yes pacman localized y/n so paru should do the samep can you make this just he instead of de_DE though to copy pacman?

daniel-seiler commented on po/de_DE.po:
```diff
@@ -0,0 +1,706 @@
+
+#: src/completion.rs:65
+msgid "could not update aur cache"
+msgstr "AUR-Cache konnte nicht geupdated werden"
```
```suggestion
msgstr "AUR-Cache konnte nicht aktualisiert werden"
```

daniel-seiler commented on po/de_DE.po:
```diff
@@ -0,0 +1,706 @@
+#: src/config.rs:961 src/info.rs:56 src/info.rs:95 src/install.rs:1755
+#: src/install.rs:1804
+msgid "Repository"
+msgstr "Repository"
```
In the [pacman repo,](https://gitlab.archlinux.org/pacman/pacman/-/blob/master/src/pacman/po/de.po) "Repository" translates to "Repositorium". Sounds unnatural, but might be worth a thought.

daniel-seiler commented on po/de_DE.po:
```diff
@@ -0,0 +1,706 @@
+
+#: src/repo.rs:201
+msgid "  nothing to do"
+msgstr "Nichts zu tun"
```
Because of the spaces in front, the string might be used as continuation of a sentence. In this case:
```suggestion
msgstr " nichts zu tun"
```
would be appropriate.

Morganamilo replied:
This still needs to be fixed.

mocdaniel replied:
Done, the double whitespace slipped my attention.

daniel-seiler reviewed (commented):
To sum up:
- IMO the technical terms are understandable (unless the r/ich_iel approach is desired)
- For some strings it isn't clear if they're used as a continuation of a sentence. If that's the case, some German strings need to start lower case.
Otherwise I liked the translation.

Morganamilo commented:
Looking through you seem to be Capitalising everything. Please Keep the case of the original message.

mocdaniel commented:
I added the suggestions by @daniel-seiler and reduced capitalisation where possible. Due to German syntax, many of the translations start with nouns which are always capitalised in German. I also changed the filename, to `de.po`.

Morganamilo commented:
> Due to German syntax, many of the translations start with nouns which are always capitalised in German.

Oh that's fine in that case.
